### PR TITLE
Support group authorization for ldap

### PR DESCRIPTION
**Pull request by mougams**
When group is provided to spoa agent by HAProxy, use it in user filter query to look up for the user in ldap.
The user filter also has to be updated to contain the string "{group}" in the user filter.

### CODE REVIEW

File: internal/auth/authenticator_ldap.go [hunk 1]
```diff
@@ -110,9 +112,20 @@ func (la *LDAPAuthenticator) Authenticate(msg *spoe.Message) (bool, []spoe.Actio
...
+			group, ok = arg.Value.(string)
+			if !ok {
+				group = ""
+			}
+			is_group_provided = true
 		}
 	}
 
+	if is_group_provided {
+		logrus.Debug(fmt.Sprintf("Group header is <%s>", group))
```
Comment (dclaisse) [inline internal/auth/authenticator_ldap.go:126]: Not a header

File: internal/auth/authenticator_ldap.go [hunk 2]
```diff
@@ -124,11 +137,15 @@ func (la *LDAPAuthenticator) Authenticate(msg *spoe.Message) (bool, []spoe.Actio
...
 	}
 
-	err = verifyCredentials(&la.connectionDetails, username, password)
+	err = verifyCredentials(&la.connectionDetails, username, password, group)
 
 	if err != nil {
 		if err == ErrUserDoesntExist {
-			logrus.Debugf("user %s does not exist", username)
+			if is_group_provided {
+				logrus.Debugf("user <%s> from group <%s> does not exist", username, group)
```
Comment (dclaisse) [inline internal/auth/authenticator_ldap.go:145]: A bit ambiguous, we should tell somehting like "user doesn't exist, or is not member of group x".

**Review by dclaisse — CHANGES_REQUESTED** [review]

File: internal/auth/authenticator_ldap.go [hunk 3]
```diff
@@ -124,11 +137,15 @@ func (la *LDAPAuthenticator) Authenticate(msg *spoe.Message) (bool, []spoe.Actio
 		return false, nil, fmt.Errorf("unable to parse basic auth header")
 	}
 
-	err = verifyCredentials(&la.connectionDetails, username, password)
+	err = verifyCredentials(&la.connectionDetails, username, password, group)
 
 	if err != nil {
 		if err == ErrUserDoesntExist {
-			logrus.Debugf("user %s does not exist", username)
+			if is_group_provided {
```
Comment (fdomain) [inline internal/auth/authenticator_ldap.go:144]: style: use camel case format isGroupProvided

File: internal/auth/authenticator_ldap.go (hunk 1 above)
Comment (mougams) [reply internal/auth/authenticator_ldap.go]: Removed - done

File: internal/auth/authenticator_ldap.go (hunk 2 above)
Comment (mougams) [reply internal/auth/authenticator_ldap.go]: Updated

File: internal/auth/authenticator_ldap.go (hunk 3 above)
Comment (mougams) [reply internal/auth/authenticator_ldap.go]: Done

**Review by dclaisse — APPROVED** [review]

**Pull request merged**